packages/core/src/components/layout/layout.ts

```diff
@@ -0,0 +1,213 @@
+// Internal Imports
+import { Component } from "../component";
+import { LayoutDirection, LayoutGrowth, LayoutComponentChild, LayoutConfigs } from "../../interfaces/index";
+import { Tools } from "../../tools";
+import { DOMUtils } from "../../services";
+import { ChartModel } from "../../model";
+
+// D3 Imports
+import { select } from "d3-selection";
+import { hierarchy, treemap, treemapSlice, treemapDice } from "d3-hierarchy";
+
+// TODO - What if there is no "growth" object?
+export class LayoutComponent extends Component {
+	// Give every layout component a distinct ID
+	// so they don't interfere when querying elements
+	static instanceCount = Math.floor(Math.random() * 99999999999);
+
+	type = "layout";
+
+	children: Array<LayoutComponentChild>;
```

Comment:
Generally we'll want to prefer types like `LayoutComponentChild[]` (i.e. "real" concrete types) ... see https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html for specifics